### PR TITLE
new(driverkit/config): add build config for 6.1.0-rpi7-rpi-2712_1

### PR DESCRIPTION
**Pull request by Nabsku**

### CODE REVIEW

**Comment by FedeDP** [comment]
Hi! Thanks for opening this PR!
Can you share `uname -a` on your debian rpi machine?

**Comment by Nabsku** [comment]
Of course! 

It's a RaspiOS Debian running on a Pi5.

`Linux node5 6.1.0-rpi7-rpi-2712 #1 SMP PREEMPT Debian 1:6.1.63-1+rpt1 (2023-11-24) aarch64 GNU/Linux`

**Comment by Nabsku** [comment]
Am not the most proficient kernel guru but I can try and find these kernelurls, if that helps? 

But it seems we've uncovered a bug in the crawler then?

**Comment by FedeDP** [comment]
> But it seems we've uncovered a bug in the crawler then?

Exactly :) 

> Am not the most proficient kernel guru but I can try and find these kernelurls, if that helps?

Yes that would be great! If you check other debian configs in the same folders, you'll see how the kernelurls are added!

**Comment by Nabsku** [comment]
I believe/hope this is correct.

Now that I think of it, does the crawler look for kernels in the Raspberry repos?

File: driverkit/config/4.0.0+driver/aarch64/debian_6.1.0-rpi7-rpi-2712_1.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+kernelversion: "1"
+kernelrelease: 6.1.0-rpi7-rpi-2712
+target: debian
+architecture: arm64
+output:
+    module: output/4.0.0+driver/aarch64/falco_debian_6.1.0-rpi7-rpi-2712_1.ko
+    probe: output/4.0.0+driver/aarch64/falco_debian_6.1.0-rpi7-rpi-2712_1.o
+kernelurls:
+  - http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-image-6.1.0-rpi7-rpi-2712_6.1.63-1+rpt1_arm64.deb
```
Comment (FedeDP) [inline driverkit/config/4.0.0+driver/aarch64/debian_6.1.0-rpi7-rpi-2712_1.yaml:9]: We normally have 3 URLs for each kernel header; example:
```
kernelurls:
    - http://mirrors.edge.kernel.org/debian/pool/main/l/linux/linux-headers-6.5.0-0.deb12.4-amd64_6.5.10-1~bpo12+1_amd64.deb
    - http://mirrors.edge.kernel.org/debian/pool/main/l/linux/linux-headers-6.5.0-0.deb12.4-common_6.5.10-1~bpo12+1_all.deb
    - http://mirrors.edge.kernel.org/debian/pool/main/l/linux/linux-kbuild-6.5.0-0.deb12.4_6.5.10-1~bpo12+1_amd64.deb
```

We have a `kbuild`, a `common` and a flavor specific headers package.
In you case, these should be:
```
- http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi7-common-rpi_6.1.63-1+rpt1_all.deb
- http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi7-rpi-2712_6.1.63-1+rpt1_arm64.deb
- http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-kbuild-6.1_6.1.63-1+rpt1_arm64.deb
```

**Comment by FedeDP** [comment]
> Now that I think of it, does the crawler look for kernels in the Raspberry repos?

I don't think so, that's the problem :D

File: driverkit/config/4.0.0+driver/aarch64/debian_6.1.0-rpi7-rpi-2712_1.yaml (hunk 1 above)
Comment (Nabsku) [reply driverkit/config/4.0.0+driver/aarch64/debian_6.1.0-rpi7-rpi-2712_1.yaml]: Ahh, guess I got confused by a file that only had the kbuild one in it and figured I'll just need the normal kernel image. Sorry!

Comment (FedeDP) [reply driverkit/config/4.0.0+driver/aarch64/debian_6.1.0-rpi7-rpi-2712_1.yaml]: No problem! I am trying to fix that issue too!

**Comment by FedeDP** [comment]
Yay:
```
Building dependency tree  [####################################]  100%
{
  "debian": [
    {
      "kernelversion": "1",
      "kernelrelease": "1:6.1.21-1+rpt1-rpi-arm64",
      "target": "debian",
      "headers": [
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-kbuild-6.1_6.1.63-1+rpt1_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi1-rpi-v8_6.1.21-1+rpt1_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi1-common-rpi_6.1.21-1+rpt1_all.deb"
      ]
    },
    {
      "kernelversion": "1",
      "kernelrelease": "1:6.1.47-1+rpt2-rpi-arm64",
      "target": "debian",
      "headers": [
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi2-common-rpi_6.1.47-1+rpt2_all.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi2-rpi-v8_6.1.47-1+rpt2_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-kbuild-6.1_6.1.63-1+rpt1_arm64.deb"
      ]
    },
    {
      "kernelversion": "1",
      "kernelrelease": "1:6.1.47-1+rpt4-rpi-arm64",
      "target": "debian",
      "headers": [
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi3-common-rpi_6.1.47-1+rpt4_all.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi3-rpi-2712_6.1.47-1+rpt4_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi3-rpi-v8_6.1.47-1+rpt4_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-kbuild-6.1_6.1.63-1+rpt1_arm64.deb"
      ]
    },
    {
      "kernelversion": "1",
      "kernelrelease": "1:6.1.54-1+rpt2-rpi-arm64",
      "target": "debian",
      "headers": [
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi4-rpi-v8_6.1.54-1+rpt2_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi4-rpi-2712_6.1.54-1+rpt2_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-kbuild-6.1_6.1.63-1+rpt1_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi4-common-rpi_6.1.54-1+rpt2_all.deb"
      ]
    },
    {
      "kernelversion": "1",
      "kernelrelease": "1:6.1.58-1+rpt2-rpi-arm64",
      "target": "debian",
      "headers": [
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi6-rpi-2712_6.1.58-1+rpt2_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi6-rpi-v8_6.1.58-1+rpt2_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-kbuild-6.1_6.1.63-1+rpt1_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi6-common-rpi_6.1.58-1+rpt2_all.deb"
      ]
    },
    {
      "kernelversion": "1",
      "kernelrelease": "1:6.1.63-1+rpt1-rpi-arm64",
      "target": "debian",
      "headers": [
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi7-rpi-v8_6.1.63-1+rpt1_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-kbuild-6.1_6.1.63-1+rpt1_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi7-rpi-2712_6.1.63-1+rpt1_arm64.deb",
        "http://archive.raspberrypi.com/debian/pool/main/l/linux/linux-headers-6.1.0-rpi7-common-rpi_6.1.63-1+rpt1_all.deb"
      ]
    }
  ]
}
```

**Comment by Nabsku** [comment]
Awesome!

**Comment by FedeDP** [comment]
If we are quick enough to merge the PR, i would let the next kernel-crawler pipeline run (tomorrow morning) create the new driverkit configs for us! Wdyt?

**Comment by Nabsku** [comment]
I'm all for it :) Anything left to do?

**Comment by FedeDP** [comment]
> I'm all for it :) Is there anything left to do? If so just say the word!

Nothing, as soon as my kenrel-crawler patch gets merged, i will tag a new release and tomorrow morning you will find Falco drivers built and published on `download.falco.org`.

**Comment by Nabsku** [comment]
Shall I close this PR? Probably not needed anymore, right? :) Thanks for your help!

**Comment by FedeDP** [comment]
I think so! You are welcome, thanks for spotting the issue!

**Comment by FedeDP** [comment]
Tag created: https://github.com/falcosecurity/kernel-crawler/releases/tag/0.14.0

**Comment by FedeDP** [comment]
Oh I didn't notice the "1:" prefix in the output kernel release:
> 1:6.1.63-1+rpt1-rpi-arm64

Will need to fix this on Monday; sorry!

**Comment by FedeDP** [comment]
Ok fixed, and currently re running kernel-crawler: https://github.com/falcosecurity/kernel-crawler/actions/runs/7246238172.
Expect drivers to come alive in ~1h!

**Comment by FedeDP** [comment]
Unfortunately, it seems driverkit is not able to build the drivers for rpi debian kernels.
Here is the fix: https://github.com/falcosecurity/driverkit/pull/314

**Comment by FedeDP** [comment]
We finally have prebuilt drivers for `rpi` debian kernels :smile: 
Thanks for your patience @Nabsku

**Pull request closed without merging**